### ISSUE TITLE
Easiest way to use conan-boost with pyenv?

### ISSUE THREAD

**Issue by colatkinson** [conan-community/community#126]
### Description of Problem, Request, or Question

I'm currently using conan-boost with `boost_python` to generate bindings for a C++ project using cmake. System-wide, I have Python 3.7 installed. However, I would like to use a `pyenv` version of Python, namely `3.6.7`.

Short of forking the repo and customizing the jamfile manually, is there any other way to set the Python path through e.g. environment/cmake variables? I'm not opposed to this, but it would be nice not to have to do so.

### Package Details (Include if Applicable)
* Package Name/Version: **boost/1.68.0**
* Operating System: **Linux**
* Operation System Version: **Arch**
* Compiler+version: **gcc-8.2.1**
* Conan version: **conan 1.10.2**
* Python version: **python 3.7.0 (system), python 3.6.7 (pyenv)**

### Steps to reproduce (Include if Applicable)

1. Set up a system with system-wide Python other than 3.6
2. Set up a minimal conan/CMake build
3. Include boost/1.68.0 and set `without_python=False`
4. Add `find_package(Boost 1.68.0 REQUIRE COMPONENTS python36)` to the CMake
5. Build will fail, being unable to find `boost_python36`

### Build logs (Include if Available)

The log output is quite long, as boost is being built. I can include the full output if it would be helpful, though. The exact error on failure is:

```CMake Error at /usr/share/cmake-3.13/Modules/FindBoost.cmake:2100 (message):
  Unable to find the requested Boost libraries.

  Boost version: 1.68.0

  Boost include path:
  /home/colin/.conan/data/boost/1.68.0/conan/stable/package/136196183e2d3946656606ca1fa63c3dd665b063/include


  Could not find the following Boost libraries:

          boost_python36

  No Boost libraries were found.  You may need to set BOOST_LIBRARYDIR to the
  directory containing Boost libraries or BOOST_ROOT to the location of
  Boost.
Call Stack (most recent call first):
  python/CMakeLists.txt:6 (find_package)
```

And during `python-cfg`, boost finds:

```notice: [python-cfg] Configuring python...
notice: [python-cfg]   user-specified version: "3.7"
notice: [python-cfg]   user-specified cmd-or-prefix: "/usr/bin/python"
notice: [python-cfg] Checking interpreter command "/usr/bin/python"...
notice: [python-cfg] running command '/usr/bin/python -c "from sys import *; print('version=%d.%d\nplatform=%s\nbase_prefix=%s\nbase_exec_prefix=%s\nexecutable=%s' % (version_info[0],version_i
nfo[1],platform,base_prefix,base_exec_prefix,executable))" 2>&1'
notice: [python-cfg] ...requested configuration matched!
notice: [python-cfg] Details of this Python configuration:
notice: [python-cfg]   interpreter command: "/usr/bin/python"
notice: [python-cfg]   include path: "/usr/include/python3.7"
notice: [python-cfg]   library path: "/usr/lib/python3.7/config" "/usr/lib"
notice: [python-cfg] Checking for NumPy...
notice: [python-cfg] running command '/usr/bin/python -c "import sys; sys.stderr = sys.stdout; import numpy; print(numpy.get_include())"'
notice: [python-cfg] NumPy disabled. Reason:
notice: [python-cfg]   /usr/bin/python -c "import sys; sys.stderr = sys.stdout; import numpy; print(numpy.get_include())" aborted with 
notice: [python-cfg]   Traceback (most recent call last):```

**Comment by SSE4** [comment]
@colatkinson I think I have fixed compatibility issues, please give an another try (branch 1.69.0)
there are options added to use custom python version (`python_executable` and `python_version`), you may try to use them.

**Comment by colatkinson** [comment]
@SSE4 Awesome, that seems like it'll be super helpful. I'll give it a shot when I have some time.

Were these changes pushed to bintray, or should I reference the repo directly?

**Comment by SSE4** [comment]
@colatkinson changes are already merged to `release/1.69.0` branch : https://github.com/conan-community/conan-boost/commit/538ed76fa05877d766c5254cdb6370e90a1c5803
new packages for 1.69.0 are already available on bintray
if you need back-porting for older versions, please lemme know

**Comment by SSE4** [comment]
@colatkinson  I am closing this one for now, as required changes were applied. please open new issue, if you still have a problem.
